classes.rb

```diff
@@ -0,0 +1,47 @@
+class Person
+  def initialize(id, name, age, parent_permission)
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+  attr_accessor :name, :age
+  attr_reader :parent_permission
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def can_use_services?
+    of_age? == true
+  end
+end
+
+class Student < Person
+  def initialize(id, name, age, parent_permission, classroom)
+    super(id, name, age, parent_permission)
+    @classroom = classroom
```

Comment:
- [ ] If you adopt my suggestion about generating IDs without having to pass them as an argument, you will update this constructor and the `super(....)` method.
- [ ] Secondly, `name` and `parent_permission` should have a default value of `unknown` and `true` respectively like so 👇🏾 

```ruby
def initialize(age, classroom, name = "Unknown", parent_permission = true)
```

Reply:
well resolved! ✔️